### ISSUE TITLE
[bug] Weird side effects when using `{/}` substitution at the edges of args under Windows

### ISSUE THREAD

**Issue by webknjaz** [tox-dev/tox#3330]
Basically, given the following config

```ini
[testenv]
commands = some-script.py path{/} {posargs} --more-args
```
or even
```ini
[testenv]
commands = 
  some-script.py \
    path{/} \
    {posargs} \
    --more-args
```

tox converts `{/}` into `\` early and then treats it as if it's typed into the config by the end-user and is an escaping character.

I was very puzzled to see that the program I'm invoking via `tox r -- few more args` was getting `path few` as a single argument (`sys.args` was getting something like `["path few", "more", "args"]`). And only on Windows :exploding_head:! But after staring at it long enough, it occurred to me that it's just an escape character.

P.S. I'm using `path{/}` because I have a habit of visually separating directory and file path values by that trailing slash. Of course, the workaround is to drop it, but I believe it reveals a deeper issue here. I haven't attempted doing any debugging, but still wanted to document this magical behavior.

**Comment by webknjaz** [comment]
Interestingly, in the place where I got `'{toxinidir}{/}requirements{/}'`, it seems to work. So quotes to the rescue I guess...

**Comment by gaborbernat** [comment]
Yeah, because is the trailing `\` the issue, so a quote sidesteps the problem. Perhaps the line collapse needs to happen before any substitution and none afterward.

**Comment by gaborbernat** [comment]
Duplicate of #3222 — same root cause: trailing `{/}` on Windows produces `\` which is then treated as an escape character, breaking command argument quoting.
